builtins/match.go

```diff
@@ -30,6 +30,15 @@ func match(a, b runtime.Object) (bool, error) {
 		default:
 			return matchRecordOf(a.(*runtime.List), b)
 		}
+	case *runtime.String:
+		if c := a.(*runtime.String); c.Len() == 1 && b.Len() == 1 {
+			if b.Value[0] == rune('?') || b.Value[0] == rune('*') || b.Value[0] == c.Value[0] {
+				return true, nil
+			}
+			return false, runtime.Errorf("tried to match %d with %d", c.Value[0], b.Value[0])
+
+		}
+		return matchRecordOf(a.(*runtime.String), b)
```

Comment:
I am not sure, but I think you can drop the type assertion (`.(*runtime.String)`).

Reply:
You can't